src/javascript/binary/base/page.js

```diff
@@ -787,7 +799,15 @@ Page.prototype = {
         var that = this;
         $('#language_select').on('change', 'select', function() {
             var language = $(this).find('option:selected').attr('class');
-            document.location = that.url_for_language(language);
+            var loc = document.location; // quick access
```

Comment:
Hi Raunak, The url_for when used returned the url and the language of the current page. When used it would return something like 

```/trading?l=EN```  even when the language is changed to a new language from the drop down . This is because unlike the ```url_for_language``` which  accept the selected language as a parameter, the url_for does not accept the selected language and return the language of the existing page before post back as a result. I thought i had to made some change to the url_for if i want it to accomodate my requirement, which add the moment i noticed is being referenced in many places.